# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [2]:
library(tidyverse) 

-- Attaching packages --------------------------------------- tidyverse 1.2.1 --
v ggplot2 2.2.1     v purrr   0.2.4
v tibble  1.4.1     v dplyr   0.7.4
v tidyr   0.7.2     v stringr 1.2.0
v readr   1.1.1     v forcats 0.2.0
-- Conflicts ------------------------------------------ tidyverse_conflicts() --
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()


# Assignment 1

First, we generate some random data:

In [3]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1         V2        V3         V4         V5          V6        V7       
1 0.09377846 0.3713894 0.01419035 0.63143487 0.001575611 0.6191592 0.3409853
2 0.49006342 0.6940533 0.63888097 0.04665147 0.324189333 0.9834746 0.5470567
  V8        V9        V10       V11        V12       V13       V14      
1 0.8014591 0.2805443 0.7531603 0.08453701 0.1912709 0.7545975 0.3367849
2 0.6191143 0.5641095 0.7297075 0.96021795 0.3483690 0.9499524 0.5311688
  V15       V16       V17       V18       V19       V20      
1 0.9337973 0.8722969 0.3572477 0.4915825 0.1852786 0.6292793
2 0.8251394 0.1089536 0.7371497 0.7777131 0.7825904 0.7153348

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [6]:
df2 <- df1 %>%
  gather(Key, Value)

head(df2)

Key Value     
1 V1  0.09377846
2 V1  0.49006342
3 V2  0.37138938
4 V2  0.69405326
5 V3  0.01419035
6 V3  0.63888097

## Assignment 2

We will generate some data:

In [7]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [8]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [10]:
df5 <- full_join(df3, df4, by = "id")
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook